src/parser/index.ts

```diff
@@ -67,87 +56,89 @@ export function parse(
         return parseLogicalAndOrExpr(ast.pop() as T.Expr);
       }
 
-      ParserError(`Unhandled keyword token with value \`${curTok.value}\``);
+      ParserError(`Unhandled keyword token with value \`${tt.current.value}\``);
     }
 
-    if (curTok.type === 'number') {
-      return parseNumberLiteral(curTok, nextToken, prevExpr);
+    if (tt.is('number')) {
+      return parseNumberLiteral(tt, prevExpr);
     }
 
-    if (curTok.type === 'string') {
-      return parseStringLiteral(curTok, nextToken, prevExpr);
+    if (tt.is('string')) {
+      return parseStringLiteral(tt, prevExpr);
     }
 
-    if (curTok.type === 'boolean') {
-      return parseBooleanLiteral(curTok, nextToken, prevExpr);
+    if (tt.is('boolean')) {
+      return parseBooleanLiteral(tt, prevExpr);
     }
 
-    if (curTok.type === 'null') {
-      return parseNullLiteral(curTok, nextToken, prevExpr);
+    if (tt.is('null')) {
+      return parseNullLiteral(tt, prevExpr);
     }
 
-    if (curTok.type === 'ident') {
-      return parseLiteral(curTok, nextToken, currentToken, parseExpr, scope, prevExpr);
+    if (tt.is('ident')) {
+      return parseLiteral(tt, parseExpr, scope, prevExpr);
     }
 
-    if (curTok.type === 'lparen') {
-      return parsePrioritizedExpr(curTok, nextToken, parseExpr, scope, prevExpr);
+    if (tt.is('lparen')) {
+      return parsePrioritizedExpr(tt, parseExpr, scope, prevExpr);
     }
 
-    if (curTok.type === 'eq') {
+    if (tt.is('eq')) {
       let resultExpr: T.Expr;
-      [curTok, resultExpr] = parseAssignmentExpr(curTok, nextToken, parseExpr, scope, ast.pop() as T.Expr);
+      resultExpr = parseAssignmentExpr(tt, parseExpr, scope, ast.pop() as T.Expr);
       return resultExpr;
     }
 
-    if (BuiltinBinaryOperators.has(curTok.value)) {
+    if (BuiltinBinaryOperators.has(tt.current.value)) {
```

Comment:
Since `tt.current.value` is often used, maybe use the accessor `tt.value` to shortcut the `tt.current.value`